ember/ember.d.ts

```diff
@@ -1765,9 +1788,43 @@ declare module Ember {
         toArray(): any[];
     }
     module RSVP {
+        interface IPromiseFunction {
+            (value: any):void;
+        }
+
+        interface IPromiseResolverFunction {
+            (resolve: IPromiseFunction, reject: IPromiseFunction): void;
+        }
+
         class Promise {
-            constructor(resolver: Function, label?: string);
+
+            /**
+              Promise objects represent the eventual result of an asynchronous operation. The
+              primary way of interacting with a promise is through its `then` method, which
+              registers callbacks to receive either a promise's eventual value or the reason
+              why the promise cannot be fulfilled.
+              @class RSVP.Promise
+              @param {function} resolver
+              @param {String} label optional string for labeling the promise.
+              Useful for tooling.
+              @constructor
+            */
+            constructor(resolver: IPromiseResolverFunction|Function, label?: string);
+
+            /**
+              The primary way of interacting with a promise is through its `then` method,
+              which registers callbacks to receive either a promise's eventual value or the
+              reason why the promise cannot be fulfilled.
+              @method then
+              @param {Function} onFulfilled
+              @param {Function} onRejected
+              @param {String} label optional string for labeling the promise.
+              Useful for tooling.
+              @return {Promise}
+            */
             then(done?: Function, fail?: Function): Promise;
+
+            // Unfortunately, adding 'catch' and 'finally' method definitions breaks type compatibility with JQuery's 'JQueryGenericPromise'

```

Comment:
Ah.. I think I figured this out.. I had told TypeScript that a method in my application code returned an Ember.RSVP.Promise, but inside the method I was calling Ember.$.getJSON which returns a jQuery promise.